rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/rename/Fields.rsc

```diff
@@ -31,19 +31,48 @@ extend framework::Rename;
 import lang::rascal::lsp::refactor::rename::Common;
 import lang::rascalcore::check::BasicRascalConfig;
 
+import lang::rascal::lsp::refactor::rename::Constructors;
+import lang::rascal::lsp::refactor::rename::Types;
+
 import lang::rascal::\syntax::Rascal;
 import analysis::typepal::TModel;
 
 import util::Maybe;
 
-// set[Define] findAdditionalDefinitions(set[Define] cursorDefs:{<_, _, _, constructorId(), _, _>, *_}, Tree _, TModel tm, Renamer r) {
-    // Find the ADT that this constructor is part of
-    // Find overloads of this ADT
-    // Find all fields with the same name in these ADT definitions
-// }
+bool isFieldRole(IdRole role) = role in {fieldId(), keywordFieldId()};
+
+set[Define] findAdditionalDefinitions(set[Define] cursorDefs:{<_, _, _, role, _, _>, *_}, Tree tr, TModel tm, Renamer r) {
+    if (!isFieldRole(role)) fail findAdditionalDefinitions;
+    if ({str fieldName} := cursorDefs.id) {
+        adtDefs = {tm.definitions[d] | loc d <- (tm.defines<idRole, defined, defined>)[dataId(), cursorDefs.scope]};
+        adtDefs += findAdditionalDefinitions(adtDefs, tr, tm, r);
+
+        // Find all fields with the same name in these ADT definitions
+        return flatMapPerFile(adtDefs, set[Define](loc f, set[Define] fileDefs) {
+            fileTr = r.getConfig().parseLoc(f);
+            fileTm = r.getConfig().tmodelForTree(fileTr);
```

Comment:
we might wanna add a shorthand in config to skip the tree and go from loc to tmodel (while internally we go via tree?)

Reply:
Have been considering it as well. Now added!